### ISSUE TITLE
client.upload_collection() API change suggestion

### ISSUE THREAD

**Issue by raulcarlomagno** [qdrant/qdrant_client#63]
One of the latest and useful change to this method was the possibility to use an Iterable for vectors argument.

But there is an scenario where it is not possible to take full advantage of this. For example, suppose we have our vectors and payload stored on mongodb. We can create a generator based on a mongodb cursor to feed the vectors, but as the API is designed today, we have to use another generator (cursor) for the payload and even another one (cursor) if we want to have a custom id.

So we will be increasing mongodb server roundtrips 3X.
if we could call upload_collection() with just a parameter, not with 3 (vectors, payload and ids), for example called 'data', one generator (cursor) will be enough.

```
def data_generator():
    for doc in mongo_cursor:
	yield {
	    'vector': doc['vector'],
	    'payload': {
		'age': doc['age'],
		'name': doc['name'],
	    },
	    'id': doc['id']
	}


client.upload_collection(
    collection_name=COLLECTION_NAME,
    data=data_generator(),
    parallel=4
)
```

**Comment by generall** [comment]
Hi @raulcarlomagno, can you please try with `qdrant-client==0.9.3`?

**Comment by raulcarlomagno** [comment]
> Hi @raulcarlomagno, can you please try with `qdrant-client==0.9.3`?

hi @generall !! but this version as far as i can check through the commits resolves the grpc jupyter issue"fix nested asyncio for jupyter" right?
it does nothing related to client.upload_collection()

**Comment by raulcarlomagno** [comment]
btw
grpc now does work on jupyter!!
thank you

**Comment by raulcarlomagno** [comment]
@generall 
what do you think about this?
https://github.com/qdrant/qdrant_client/issues/63#issue-1349364716
is it hard to implement it?

**Comment by generall** [comment]
no, should not be that hard. The question is how to make the interface compatible. Will look into it

**Comment by generall** [comment]
@raulcarlomagno check out a new `upload_records` method - https://github.com/qdrant/qdrant_client/blob/master/qdrant_client/qdrant_client.py#L1110, does it solves your issue?

Here is also an example of how to use it https://github.com/qdrant/qdrant_client/blob/master/tests/test_qdrant_client.py#L48

Availabe in 0.9.4

**Comment by raulcarlomagno** [comment]
> @raulcarlomagno check out a new `upload_records` method - https://github.com/qdrant/qdrant_client/blob/master/qdrant_client/qdrant_client.py#L1110, does it solves your issue?
> 
> Here is also an example of how to use it https://github.com/qdrant/qdrant_client/blob/master/tests/test_qdrant_client.py#L48
> 
> Availabe in 0.9.4

@generall 
It worked flawlessly!!

I think that progressbar should be optional, it could bother inside airflow logs for example, i dont know how it will be printed in logs

**Comment by generall** [comment]
> I think that progressbar should be optional, it could bother inside airflow logs for example, i dont know how it will be printed in logs

Makes sense!
Here is PR: https://github.com/qdrant/qdrant_client/pull/65

**Comment by imneonizer** [comment]
I have to use below snippet to work with grpc inside jupyter notebook with `qdrant-client==0.9.0`:
````python
import nest_asyncio
nest_asyncio.apply()
````

Is it fixed in `qdrant-client==0.9.3`
